Sources/Plasma/PubUtilLib/plNetGameLib/Private/plNglAuth.cpp

```diff
@@ -1861,7 +1861,7 @@ bool RecvMsg<Auth2Cli_ServerAddr>(const uint8_t in[], unsigned, void*)
     hsLockGuard(s_critsect);
     if (s_active) {
         s_active->token = msg.token;
-        s_active->addr.SetHost(msg.srvAddr);
+        s_active->addr.SetHost(hsToBE32(msg.srvAddr));
```

Comment:
I'm guessing we need to byteswap here because the server is (incorrectly) sending the address in LE rather than network order?

Reply:
Well, it's "correct" in the sense that it's consistent with the rest of the auth server protocol - IPv4 addresses are treated as integers, so they are transmitted little-endian. It's definitely a weird choice though.

An extra twist is that the NetCli receive machinery also does an implicit endianness conversion before the message gets here. So the IPv4 address is received little-endian, then implicitly converted to host byte order, and finally explicitly converted to big-endian. Fun!